tests/utils/shopper.js

```diff
@@ -47,6 +58,148 @@ export const shopper = {
+		await expect( page.title() ).resolves.toMatch( productname );
+		await expect( page ).toMatchElement( 'h1.entry-title', productname );
+	},
+
+	// The wcShopper.emptyCart is broken. This one works with the Cart block for extra snaz
+	emptyCart: async () => {
+		await shopper.goToCartBlock();
+
+		// Remove coupons
+		let couponRemoveLinks = await page.$$(
+			'.wc-block-components-chip__remove'
+		);
+		while ( couponRemoveLinks.length > 0 ) {
+			await couponRemoveLinks[ 0 ].click();
+			couponRemoveLinks = await page.$$(
+				'.wc-block-cart-item__remove-link'
+			);
+		}
+
+		// Remove products if they exist
+		let productRemoveLinks = await page.$$(
+			'.wc-block-cart-item__remove-link'
+		);
+		while ( productRemoveLinks.length > 0 ) {
+			await productRemoveLinks[ 0 ].click();
+			productRemoveLinks = await page.$$(
+				'.wc-block-cart-item__remove-link'
+			);
+		}
+		await page.waitForSelector( '.wp-block-woocommerce-empty-cart-block' );
+	},
+
+	addCoupon: async ( couponCode ) => {
+		const title = await page.title();
+		if ( ! title.includes( 'Cart Block' ) ) {
+			await shopper.goToCartBlock();
+		}
+		// Make sure the coupon panel is open
+		const applyButton = await page.$(
+			'.wc-block-components-totals-coupon__button'
+		);
+		if ( ! applyButton ) {
+			await page.click( '.wc-block-components-panel__button' );
+		}
+		await page.type(
+			'.wc-block-components-totals-coupon__input input',
+			couponCode
+		);
+		await page.click( '.wc-block-components-totals-coupon__button' );
+		await page.waitForSelector( '.wc-block-components-chip__text', {
+			text: couponCode,
+		} );
+	},
+
+	fillInCheckoutWithTestData: async () => {
+		const shippingOrBilling = ( await page.$( '#shipping-first_name' ) )
+			? 'shipping'
+			: 'billing';
+		const testData = {
+			first_name: 'John',
+			last_name: 'Doe',
+			shipping_address_1: '123 Easy Street',
+			country: 'United States (US)',
+			city: 'New York',
+			state: 'New York',
+			postcode: '90210',
+		};
+		await shopper.fillInCheckoutAddress( testData, shippingOrBilling );
+	},
+
+	fillInCheckoutAddress: async (
+		address,
+		shippingOrBilling = 'shipping'
+	) => {
+		await shopper.clearAndTypeInput(
+			`#${ shippingOrBilling }-first_name`,
+			address.first_name
+		);
+		await shopper.clearAndTypeInput(
+			`#${ shippingOrBilling }-last_name`,
+			address.last_name
+		);
+		await shopper.clearAndTypeInput(
+			`#${ shippingOrBilling }-address_1`,
+			address.shipping_address_1
+		);
+		await shopper.clearAndTypeInput(
+			`#${ shippingOrBilling }-country input`,
+			address.country
+		);
+		await shopper.clearAndTypeInput(
+			`#${ shippingOrBilling }-city`,
+			address.city
+		);
+		await shopper.clearAndTypeInput(
+			`#${ shippingOrBilling }-state input`,
+			address.state
+		);
+		await shopper.clearAndTypeInput(
+			`#${ shippingOrBilling }-postcode`,
+			address.postcode
+		);
+	},
+
+	placeOrder: async () => {
+		await sleep( 1 );
```

Comment:
Could [uiUnblocked](https://www.npmjs.com/package/@woocommerce/e2e-utils) be used instead?

Reply:
That was my first try, I'm not sure how it works in woocommerce, when the `.blockUI` class is added on which the check is based, but it didn't work in this case. I suspect because we have to wait until the data store is updated, at which time there might be some UI changes and things can become unresponsive for some milliseconds. I know this isn't a neat solution. We could maybe try to write something more clever but I thought this was a good first pass. Happy to consider alternatives though

Reply:
I've made this suggestion based on the WC implementation. @niels also added a similar utility

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5860#discussion_r813802506

Reply:
I've made this suggestion based on the WC implementation. @niels also added a similar utility

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5860#discussion_r813802506

Reply:
I've made this suggestion based on the WC implementation. @niels also added a similar utility

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5860#discussion_r813802506